core/dbt/parser/schemas.py

```diff
@@ -351,6 +352,30 @@ def create_test_node(
         column_name: Optional[str],
     ) -> ParsedSchemaTestNode:
 
+        # TODO: (PR input needed) Should this be applied to all tests?
+        # In theory we already check non-schema tests for "unique" name uniqueness in
+        # core/dbt/contracts/graph/manifest.py @ _check_duplicates()
+
+        HASH_LENGTH = 10
+
+        if 'schema' in tags:
+            # TODO: (PR input needed) is this a complete set of uniquie-ifying data?
```

Comment:
> What do you think about just hashing together everything in test_metadata.kwargs? 
@jtcohen6 Hashing the entire test_metadata is a little tricky because there's no guarantee of order.  We can enforce an order by doing a recursive sorting/frozendict/whatever, but it _might_ be a bit of a hit performance wise.  It might not.. I have no idea how many times we'd be likely to call this in any given run, nor am I aware of the full extent of the size that `test_metadata` can reach.  I'm happy to try it out though.

> Those are both valid tests, but the only way they look different to this `unique_id` construction is in their name
From an operational point of view, I think that's fine.  as long as `unique_id` is _actually_ unique. Then everything should "just work", although I agree it's not a very helpful / human-friendly.

Reply:
I went ahead and implemented a quick semantic hashing algo (AKA sort-it-then-hash-it) so we can include the entire test_metadata dict.